add_node.rb

```diff
@@ -1,17 +1,67 @@
 #!/usr/bin/env ruby
 
 require 'json'
+require 'date'
+
+def semester_exists semester
+	possible_semesters = ["mid spring","end spring","mid autumn","end autumn"]
+	if possible_semesters.include? semester
```

Comment:
@athityakumar This was a quirk I was going to ask you to fix.

@vivekiitkgp He is returning false here, so that the `while` loop condition becomes false.

But I think we all would prefer a much more intuitive method like this:
```rb
while !semester_exists(semester)
...
```
and inside semester exists, actually return `true` if `semester` is okay, otherwise return `false`.